services/management/ia/common.go

```diff
@@ -123,6 +124,24 @@ func convertRule(l *logrus.Entry, rule *models.Rule, template templateInfo, chan
 		return nil, errors.Wrap(err, "failed to convert timestamp")
 	}
 
+	params := make(map[string]string, len(rule.Params))
+	for _, p := range rule.Params {
+		var value string
+		switch p.Type {
+		case models.Float:
+			value = fmt.Sprint(p.FloatValue)
+		case models.Bool:
+			value = fmt.Sprint(p.BoolValue)
+		case models.String:
+			value = p.StringValue
+		default:
+			l.Warnf("Invalid parameter type %s", p.Type)
+			continue
+		}
```

Comment:
Hm, seems we should not just skip such parameter, but use an empty string as value to keep parameters order. Otherwise, expression will be filled wrong.

And same in `services/management/ia/rules_service.go`

Reply:
What do you mean by parameters order? It's unordered map.
If key is missing, if it will raise error because of
```
func newParamTemplate() *template.Template {
	return template.New("").Option("missingkey=error").Delims("[[", "]]")
}
```

Reply:
Oh, I was thinking about something else (about slice for example) while writing the comment 🤦
Nevermind)